### PR TITLE
Add AT4

### PR DESCRIPTION
**Pull request by Phenomphear**
#### Summary

Add AT4
```SUMMARY: Content "Adds AT4 Launcher"```

#### Additional context
Could use help making sure AT4 spawns loaded as debug only spawns an empty one.

### CODE REVIEW

**Comment by RarkGrames** [comment]
Description should mention that it is a rocket launcher, for the layman.

**Comment by kevingranade** [comment]
Need json styling, several lines have tabs or something.

File: data/json/items/gun/84x246mm.json [hunk 1]
```diff
@@ -21,6 +21,21 @@
     "clip_size": 1,
     "reload": 300,
     "loudness": 200,
-    "valid_mod_locations": [ [ "accessories", 4 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ], [ "sling", 1 ] ]
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ] ]
+  },
+  {
+    "id": "AT4",
+    "type": "GUN",
+	  "copy-from": "LAW",
```
Comment (kevingranade) [inline data/json/items/gun/84x246mm.json]:
```suggestion
    "copy-from": "LAW",
```

File: data/json/items/gun/84x246mm.json [hunk 2]
```diff
@@ -21,6 +21,21 @@
...
+    "name": "AT4",
+    "description": "Mil-Spec Rocket Launcher. An 84-mm unguided, portable, single-shot recoilless smoothbore weapon used primarily by the US Military.",
+	  "flags": [ "NO_UNLOAD", "BACKBLAST", "NEVER_JAMS", "TRADER_AVOID", "FIRE_TWOHAND", "NO_REPAIR" ],
```
Comment (kevingranade) [inline data/json/items/gun/84x246mm.json]:
```suggestion
    "flags": [ "NO_UNLOAD", "BACKBLAST", "NEVER_JAMS", "TRADER_AVOID", "FIRE_TWOHAND", "NO_REPAIR" ],
```

File: data/json/items/gun/84x246mm.json [hunk 3]
```diff
@@ -21,6 +21,21 @@
...
+    "ammo": "84x246mm",
+    "weight": 6803,
+    "volume": 14,
+    "bashing": 4,
+	  "dispersion": 200,
```
Comment (kevingranade) [inline data/json/items/gun/84x246mm.json]:
```suggestion
    "dispersion": 200,
```

File: data/json/items/gun/84x246mm.json [hunk 4]
```diff
@@ -21,6 +21,21 @@
...
     "loudness": 200,
-    "valid_mod_locations": [ [ "accessories", 4 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ], [ "sling", 1 ] ]
+    "valid_mod_locations": [ [ "accessories", 4 ], [ "grip", 1 ], [ "mechanism", 4 ], [ "sights", 1 ] ]
+  },
+  {
+    "id": "AT4",
+    "type": "GUN",
+	  "copy-from": "LAW",
+    "name": "AT4",
+    "description": "Mil-Spec Rocket Launcher. An 84-mm unguided, portable, single-shot recoilless smoothbore weapon used primarily by the US Military.",
```
Comment (Night-Pryanik) [inline data/json/items/gun/84x246mm.json]:
```suggestion
    "description": "Mil-Spec rocket launcher.  An 84-mm unguided, portable, single-shot recoilless smoothbore weapon used primarily by the US military.",
```

File: data/json/items/gun/84x246mm.json (hunk 2 above)
Comment (Night-Pryanik) [inline data/json/items/gun/84x246mm.json]: If you want to add flags to the ones you inherited from `copy-from`, use 
`"extend": { "flags": [ "FIRE_TWOHAND", "NO_REPAIR" ] }`
instead.

**Comment by Night-Pryanik** [comment]
> Could use help making sure AT4 spawns loaded as debug only spawns an empty one.

All launchers are spawned empty. The chance for it to spawn loaded is determined in item groups, with its `"magazine": xxx` field, see `helicopter` item group in `item_groups.json` for example.

**Comment by Phenomphear** [comment]
Okay, I think I understand what you mean for making it spawn loaded. Just to make sure I have this right, I need to go through `item_groups.json` and find all the places I want it to spawn loaded and see if `"magazine": 100` is there and if not can I add it as `[ "AT4", "prob":FOO, "charges": 1 ],

Otherwise, it will either spawn loaded or not depending on that `"magazine":` line for the item group?

**Comment by Night-Pryanik** [comment]
Yeah, exactly.

**Comment by Phenomphear** [comment]
I'm happy with it just spawning in the helicopter and any other itemgroups that have the AT4's group, I forgot which one I put it in. Sometimes an AT4 might be used already so if it spawns empty, that works. If there is anything else you don't like or see, please let me know.

**Pull request merged**